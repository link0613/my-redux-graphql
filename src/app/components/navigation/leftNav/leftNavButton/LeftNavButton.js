// @flow

// #region imports
import React, { PureComponent } from 'react';
import { Link } from 'react-router-dom';
// #endregion

// #region flow types
export type Props = {
  link: string,
  label: string,
  viewName: string,
  onClick: (event: SyntheticEvent<>, viewName: string) => any,
};

export type State = { ...any };
// #endregion

class LeftNavButton extends PureComponent<Props, State> {
  // #region lifecycle
  render() {
    const { link, label } = this.props;

    return (
      <li>
        <Link to={link} onClick={this.handleLeftNavItemClick}>
          {label}
        </Link>
      </li>
    );
  }
  // #endregion

  // #region on Link click event
  handleLeftNavItemClick = (event: SyntheticEvent<>) => {
    const { onClick, viewName } = this.props;
    onClick(event, viewName);
  };
  // #endregion
}

export default LeftNavButton;
