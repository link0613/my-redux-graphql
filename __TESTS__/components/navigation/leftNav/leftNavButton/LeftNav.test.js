import React from 'react';
import LeftNavButton from '../../../../../src/app/components/navigation/leftNav/leftNavButton/LeftNavButton';
import renderer from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { MemoryRouter } from 'react-router';
import { mount } from 'enzyme';

describe('LeftNavButton (navigation) component', () => {
  it('renders as expected', () => {
    const props = {
      onClick: () => {},
      link: 'link',
      label: 'label',
      viewName: '/',
    };
    const component = renderer
      .create(
        <div>
          <MemoryRouter>
            <LeftNavButton {...props} />
          </MemoryRouter>
        </div>,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('should call onClick on a click', () => {
    const onClickMock = jest.fn();
    const props = {
      onClick: onClickMock,
      link: 'link',
      label: 'label',
      viewName: '/',
    };

    const component = mount(
      <div>
        <MemoryRouter>
          <LeftNavButton {...props} />
        </MemoryRouter>
      </div>,
    );

    const button = component.find('a');
    button.simulate('click');

    expect(onClickMock).toHaveBeenCalled();
  });
});
