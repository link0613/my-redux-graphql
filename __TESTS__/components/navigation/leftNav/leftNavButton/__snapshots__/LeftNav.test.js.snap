// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftNavButton (navigation) component renders as expected 1`] = `
<div>
  <li>
    <a
      href="link"
      onClick={[Function]}
    >
      label
    </a>
  </li>
</div>
`;
